analyses/cell-type-ewings/scripts/select-cell-types.R

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env Rscript
+
+# This script is used to generate a table of reference cells for a given ScPCA libray that are annotated by 
+# SingleR and/or CellAssign to have a specific label
+# The output is a table that contains columns for barcodes, reference cell class, 
+# and then columns for each of the original cell type annotations for that cell 
+# The `reference_cell_class` column will contain either tumor or normal 
+
+project_root <- here::here()
+renv::load(project_root)
+
+suppressPackageStartupMessages({
+  library(optparse)
+  library(SingleCellExperiment)
+})
+
+option_list <- list(
+  make_option(
+    opt_str = c("--sce_file"),
+    type = "character",
+    help = "Path to RDS file containing a processed SingleCellExperiment object from scpca-nf"
+  ),
+  make_option(
+    opt_str = c("--normal_cells"),
+    type = "character",
+    default = NULL,
+    help = "Comma separated list of cell types to pull from annotations present and save as normal cells."
+  ),
+  make_option(
+    opt_str = c("--tumor_cells"),
+    type = "character",
+    default = NULL,
+    help = "Comma separated list of cell types to pull from annotations present and save as tumor cells."
+  ),
+  make_option(
+    opt_str = c("--output_filename"),
+    type = "character",
+    help = "Full path to file to store saved table of cell barcodes."
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Set up -----------------------------------------------------------------------
+
+# make sure path to sce file exists
+stopifnot("sce_file does not exist" = file.exists(opt$sce_file))
+
+# check that either normal or tumor cell types are provided 
+stopifnot("Either normal_cells or tumor_cells must be provided" = 
+            !is.null(opt$normal_cells) || !is.null(opt$tumor_cells))
+
+# read in sce file
+sce <- readr::read_rds(opt$sce_file)
+
+# create output directory if not already present
+output_dir <- dir(opt$output_filename)
+fs::dir_create(output_dir)
+
+# Create table of reference cells ----------------------------------------------
+
+# possible annotation columns 
+annotation_columns <- c(
+  "singler_celltype_annotation",
+  "cellassign_celltype_annotation"
+)
+
+# get dataframe with barcodes and associated annotation 
+coldata_df <- colData(sce) |>
+  as.data.frame() |> 
+  dplyr::select(barcodes, all_of(annotation_columns)) |> 
+  tidyr::pivot_longer(all_of(annotation_columns),
+                      names_to = "celltype_method",
+                      values_to = "celltype")
+
+# get list of normal and tumor cell types to save as refernences 
+if(!is.null(opt$normal_cells)){
+  normal_cell_types <- stringr::str_split(opt$normal_cells, pattern = ",") |>
+    unlist()
+} else {
+  normal_cell_types <- ""
+}
+
```

Comment:
Since you are going to set these to "", you could remove a lot of logic by just making that the default value when reading in. You would only need:
```suggestion
normal_cell_types <- stringr::str_split_1(opt$normal_cells, pattern = ",") 
```

Of course, you would also have to update the check to check whether the args `== ""` instead of `is.null`, but that should be pretty easy too:

```
"Either normal_cells or tumor_cells must be provided" = 
    opt$normal_cells != "" || opt$tumor_cells != ""
```